programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1775,6 +1784,67 @@ declare_syscall!(
     }
 );
 
+declare_syscall!(
+    // Poseidon
+    SyscallPoseidonBn254,
+    fn inner_call(
+        invoke_context: &mut InvokeContext,
+        vals_addr: u64,
+        vals_len: u64,
+        result_addr: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &mut MemoryMapping,
+    ) -> Result<u64, Error> {
+        let budget = invoke_context.get_compute_budget();
+        let Some(cost) = budget.poseidon_cost.get(&(vals_len as usize)) else {
```

Comment:
If we can, instead of using a hash function, let's interpolate the costs with a quadratic function here.